domain/schema/controller.go

```diff
@@ -292,7 +292,21 @@ CREATE UNIQUE INDEX idx_controller_node_dqlite_node
 ON controller_node (dqlite_node_id);
 
 CREATE UNIQUE INDEX idx_controller_node_bind_address
-ON controller_node (bind_address);`)
+ON controller_node (bind_address);
+
+CREATE TABLE api_address (
+    uuid            TEXT PRIMARY KEY,
+    controller_id   TEXT,
+    address         TEXT,
+    agent_usage     BOOLEAN,
+    CONSTRAINT      fk_model_migration_target_controller
+        FOREIGN KEY (controller_id)
+        REFERENCES  controller_node(controller_id)
+);
+
+CREATE UNIQUE INDEX idx_api_address
+ON api_address (controller_id, address);
```

Comment:
I'm ok with this, but I have to wonder if it is the "right" way to represent it.
The 'agent_usage' is just so that we can do a query mixing mgmt_space with the spaces for the addresses on the machines.

Do we *want* to denormalize that information rather than just having a query that filters for machines that are controller machines, and the spaces that are in mgmt_space ?